include/zcbor_common.h

```diff
@@ -80,6 +80,8 @@ struct zcbor_state_constant {
 	zcbor_state_t *backup_list;
 	uint_fast32_t current_backup;
 	uint_fast32_t num_backups;
+	uint_fast8_t error;
+	bool stop_on_error;
```

Comment:
Can this be made #ifdef configurable, with three possible options:
1) not used - is not included into code, so does not increase flash code size;
2) used and run-time configurable - as is currently;
3) used and compile time enforced?

The third option would be useful for code that does not care about changing the behavior at run-time; there is an example of such code here: https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c#L145-L164 that emulates the feature with lazy evaluation of `||` operator in C, although it is not able to get real error code (just that error happened), which is possible with this proposed feature.

Reply:
OK for now but there is a case: for example mcumgr always stops processing on any error so having the flag does not matter for it, it will just have it always true.